geopandas/tools/tests/test_sjoin.py

```diff
@@ -176,6 +176,7 @@ def test_inner(self, predicate, predicate_kw, dfs):
 
         exp = expected[predicate].dropna().copy()
         exp = exp.drop("geometry_y", axis=1).rename(columns={"geometry_x": "geometry"})
+        exp = exp.set_geometry("geometry")  # cast up to a gdf
```

Comment:
Is the "cast up to gdf" still correct? I would assume that now the drop/rename retains the GeoDataFrame class, but that only the active geometry column is not set properly?